lib/reconciliationText.js

```diff
@@ -0,0 +1,198 @@
+    "reconciliation_not_necessary",
+    "can_be_reconciled_without_data",
+    "only_reconciled_with_data",
+  ];
+  static TEMPLATE_TYPE = "reconciliationText";
+  static TEMPLATE_FOLDER = fsUtils.FOLDERS[this.TEMPLATE_TYPE];
+  constructor() {}
+
+  /**
+   * Process the response provided by the Silverfin API and store every detail in its corresponding file (liquid files, config file, etc)
+   * @param {number} firmId
+   * @param {object} template
+   */
+  static async save(firmId, template) {
+    if (this.#missingHandle(template)) return;
+    if (templateUtils.missingLiquidCode(template)) return;
+
+    const handle = template.handle;
+    fsUtils.createTemplateFolders(this.TEMPLATE_TYPE, handle);
+
+    // Liquid files
+    const mainPart = template.text;
+    const textParts = templateUtils.filterParts(template);
+    fsUtils.createTemplateFiles(
+      this.TEMPLATE_TYPE,
+      handle,
+      mainPart,
+      textParts
+    );
+
+    // Liquid Test YAML
+    let testContent = "# Add your Liquid Tests here";
+    if (template.tests) {
+      testContent = template.tests;
+    }
+    fsUtils.createLiquidTestFiles(this.TEMPLATE_TYPE, handle, testContent);
+
+    // Config Json File
+    let existingConfig = fsUtils.readConfig(this.TEMPLATE_TYPE, handle);
+    const configDetails = this.#prepareConfigDetails(template);
+    const configContent = {
+      id: {
+        ...existingConfig?.id,
+        [firmId]: template.id,
+      },
+      test: `tests/${handle}_liquid_test.yml`,
+      ...configDetails,
+    };
+    fsUtils.writeConfig(this.TEMPLATE_TYPE, handle, configContent);
+  }
+
+  /**
+   * Read all necessary files and prepare the object to be sent to the Silverfin API
+   * @param {string} handle The handle of the template to read
+   * @returns {object} The object to be sent to the Silverfin API
+   */
+  static read(handle) {
+    fsUtils.createTemplateFolders(this.TEMPLATE_TYPE, handle);
+    // Config.json
+    let templateConfig = fsUtils.readConfig(this.TEMPLATE_TYPE, handle);
+    templateConfig = this.#checkHandleAndNameInConfig(templateConfig, handle);
+    let template = this.#filterConfigItems(templateConfig);
+    template = this.#checkReconciliationType(template);
+    // Liquid Tests
+    templateConfig = this.#createLiquidTest(handle, templateConfig);
+    // Liquid
+    this.#createMainLiquid(handle);
+    template.text = this.#readMainLiquid(handle, templateConfig);
+    template.text_parts = this.#readPartsLiquid(handle, templateConfig);
+
+    fsUtils.writeConfig(this.TEMPLATE_TYPE, handle, templateConfig);
+    return template;
+  }
+
+  /** Update template's id for the corresponding firm in template's config file */
+  static updateTemplateId(firmId, handle, templateId) {
+    let templateConfig = fsUtils.readConfig(this.TEMPLATE_TYPE, handle);
+    templateConfig.id[firmId] = templateId;
+    fsUtils.writeConfig(this.TEMPLATE_TYPE, handle, templateConfig);
+  }
+
+  static #missingHandle(template) {
```

Comment:
Why exactly is the `#` in front of certain static functions?

Reply:
It will mark those functions as private. So they are only intended to be called from other functions inside the class.